taipy/core/_orchestrator/_dispatcher/_job_dispatcher.py

```diff
@@ -55,30 +55,37 @@ def stop(self, wait: bool = True, timeout: Optional[float] = None):
             wait (bool): If True, the method will wait for the dispatcher to stop.
             timeout (Optional[float]): The maximum time to wait. If None, the method will wait indefinitely.
         """
-        self.stop_wait = wait
-        self.stop_timeout = timeout
         self._STOP_FLAG = True
+        if wait and self.is_alive():
+            self._logger.debug("Waiting for the dispatcher thread to stop...")
+            self.join(timeout=timeout)
```

Comment:
Moved from the run method because `join` must be called from another thread.

Reply:
should we use `self.is_running` instead of `self.is_alive`? They are the same by the way, just asking for semantic-wise

Reply:
I would go for `is_running` then, since the method name is `run`.